File: src/main/java/rosatech/api/metatileentity/RosaProgressIndicator.java
package rosatech.api.metatileentity;

import gregtech.api.gui.resources.TextureArea;
import gregtech.api.gui.widgets.ProgressWidget;

public class RosaProgressIndicator {
    public TextureArea progressBarTexture;
    public ProgressWidget.MoveType progressMoveType;

    public RosaProgressIndicator(TextureArea progressBarTexture, ProgressWidget.MoveType progressMoveType) {
        this.progressBarTexture = progressBarTexture;
        this.progressMoveType = progressMoveType;
    }
}

File: src/main/kotlin/rosatech/api/metatileentity/RosaProgressIndicator.kt
package rosatech.api.metatileentity

import gregtech.api.gui.resources.TextureArea
import gregtech.api.gui.widgets.ProgressWidget.MoveType

data class RosaProgressIndicator(
    var progressBarTexture: TextureArea,
    var progressMoveType: MoveType
)
